pkg/solana/txm/txm.go

```diff
@@ -790,6 +813,41 @@ func (txm *Txm) Enqueue(ctx context.Context, accountID string, tx *solanaGo.Tran
 	return nil
 }
 
+func (txm *Txm) waitForTxStatus(ctx context.Context, transactionID string, desiredStatus types.TransactionStatus) error {
+	waitCtx, cancel := context.WithTimeout(ctx, txm.cfg.TxTimeout())
+	defer cancel()
+
+	backoff := 1 * time.Second
+	maxBackoff := 8 * time.Second
+
+	for {
+		select {
+		case <-waitCtx.Done():
+			return fmt.Errorf("context ended while waiting for finality of transaction %s", transactionID)
+		case <-time.After(backoff):
+			status, err := txm.GetTransactionStatus(waitCtx, transactionID)
+			if err != nil {
+				return fmt.Errorf("error fetching transaction status: %w", err)
+			}
+			switch status {
+			case types.Failed, types.Fatal:
+				return fmt.Errorf("transaction %s failed", transactionID)
+			default:
+				if status >= desiredStatus {
+					// if status is equal to or greater than desired status, return
+					txm.lggr.Debugw("ATA transaction reached state", "status", status, "transactionID", transactionID)
```

Comment:
nit: think the message got carried over from the ATA code
```suggestion
					txm.lggr.Debugw("transaction reached state", "status", status, "transactionID", transactionID)
```